flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/client/HiveShimV120.java

```diff
@@ -194,4 +200,108 @@ public void makeSpecFromName(Map<String, String> partSpec, Path currPath) {
 		}
 	}
 
+	@Override
+	public ObjectInspector getObjectInspectorForConstant(PrimitiveTypeInfo primitiveTypeInfo, Object value) {
+		String className;
+		switch (primitiveTypeInfo.getPrimitiveCategory()) {
+			case BOOLEAN:
+				className = "org.apache.hadoop.hive.serde2.objectinspector.primitive.JavaConstantBooleanObjectInspector";
```

Comment:
curious why does 1.1.0 check WritableConstant and 1.1.1 check JavaConstant?

Reply:
Because java constant object inspectors are not available until 1.2.0